utils/python/CIME/buildlib.py

```diff
@@ -0,0 +1,49 @@
+from CIME.XML.standard_module_setup import *
+from CIME.case import Case
+"""
+buildlibx - given the caseroot and component name(cmp), build the stub component
+"""
+
+class buildlib(object):
+    def __init__(self, case, model, comp):
+        self.case = case
+        self.model = model
+        self.comp = comp
+        self.caseroot = case.get_value("CASEROOT")
```

Comment:
Pylint will flag that (it's good practice to initialize all your members in the constructor). I think this is probably the correct design if there are going to be multiple subclasses of buildlib.